src/OpeningHoursAlgorithms.h

```diff
@@ -30,27 +30,30 @@
 
 QVariantMap mergeElements(const QVariantList& openingHours,
                           const unsigned from,
-                          const unsigned until)
+                          const unsigned until,
+                          const int todayIndex)
 {
+    QVariantMap result;
+
     if (from == until)
     {
-        return openingHours[from].toMap();
+        result = openingHours[from].toMap();
     }
     else
     {
-        QVariantMap result;
-
         const auto& fromDay  = openingHours[from]. toMap()["day"].toString();
         const auto& untilDay = openingHours[until].toMap()["day"].toString();
 
-        result["day"]   = fromDay + " - " + untilDay;
-        result["hours"] = openingHours[from].toMap()["hours"];
-
-        return result;
+        result["day"]      = fromDay + " - " + untilDay;
+        result["hours"]    = openingHours[from].toMap()["hours"];
     }
+
+    result["current"]  = from <= todayIndex && todayIndex <= until;
```

Comment:
OpeningHoursAlgorithms.h:51: warning: comparison of integers of different signs: 'const unsigned int' and 'const int'
OpeningHoursAlgorithms.h:51: warning: comparison of integers of different signs: 'const int' and 'const unsigned int'